test/commands/extended/is_reattachable_test.py

```diff
@@ -0,0 +1,191 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, \
+  unicode_literals
+
+from unittest import TestCase
+
+import filters as f
+from filters.test import BaseFilterTestCase
+from six import text_type
+
+from iota import Address, Iota
+from iota.adapter import MockAdapter
+from iota.commands.extended.is_reattachable import IsReattachableCommand
+
+
+class IsReattachableRequestFilterTestCase(BaseFilterTestCase):
+  filter_type = IsReattachableCommand(MockAdapter()).get_request_filter
+  skip_value_check = True
+
+  # noinspection SpellCheckingInspection
+  def setUp(self):
+    super(IsReattachableRequestFilterTestCase, self).setUp()
+
+    # Define a few valid values that we can reuse across tests.
+    self.address_1 = (
+      'TESTVALUE9DONTUSEINPRODUCTION99999EKJZZT'
+      'SOGJOUNVEWLDPKGTGAOIZIPMGBLHC9LMQNHLGXGYX'
+    )
+
+    self.address_2 = (
+      'TESTVALUE9DONTUSEINPRODUCTION99999FDCDTZ'
+      'ZWLL9MYGUTLSYVSIFJ9NGALTRMCQVIIOVEQOITYTE'
+    )
+
+  def test_pass_happy_path(self):
+    """
+    Filter for list of valid string addresses
+    """
+
+    request = {
+      # Raw trytes are extracted to match the IRI's JSON protocol.
+      'addresses': [
+        self.address_1,
+        Address(self.address_2)
+      ],
+    }
+
+    filter_ = self._filter(request)
+
+    self.assertFilterPasses(filter_)
+
+    self.assertDictEqual(
+      filter_.cleaned_data,
+      {
+        'addresses': [
+          text_type(Address(self.address_1)),
+          text_type(Address(self.address_2))
+        ],
+      },
+    )
+
+  def test_pass_compatible_types(self):
+    """
+    The incoming request contains values that can be converted to the
+    expected types.
+    """
+    request = {
+      'addresses': [
+        Address(self.address_1),
+        bytearray(self.address_2.encode('ascii')),
+      ],
+    }
+
+    filter_ = self._filter(request)
+
+    self.assertFilterPasses(filter_)
+    self.assertDictEqual(
+      filter_.cleaned_data,
+      {
+        'addresses': [self.address_1, self.address_2],
+      },
+    )
+
+  def test_pass_incompatible_types(self):
+    """
+    The incoming request contains values that can NOT be converted to the
+    expected types.
+    """
+    request = {
+      'addresses': [
+        1234234,
+        False
+      ],
+    }
+
+    self.assertFilterErrors(
+      request,
+      {
+        'addresses.0': [f.Type.CODE_WRONG_TYPE],
+        'addresses.1': [f.Type.CODE_WRONG_TYPE]
+      },
+    )
+
+  def test_fail_empty(self):
+    """
+    The incoming request is empty.
+    """
+    self.assertFilterErrors(
+      {},
+      {
+        'addresses': [f.FilterMapper.CODE_MISSING_KEY],
+      },
+    )
```

Comment:
Could we include a test for `addresses` not an array (e.g., calling `is_reattachable(Address(...))`)?

Reply:
How is that possible with filters (https://filters.readthedocs.io/en/latest/#)?

I couldn't find a proper configuration for accepting both, an `Address` OR an array containing `Address` objects?

That's how validation done at the time beeing:
```
class IsReattachableRequestFilter(RequestFilter):
  def __init__(self):
    super(IsReattachableRequestFilter, self).__init__(
      {
        'addresses': (
          f.Required
          | f.Array
          | f.FilterRepeater(
            f.Required
            | Trytes(result_type=Address)
            | f.Unicode(encoding='ascii', normalize=False)
          )
        )
      }
    )
```

Reply:
Sorry; I should have clarified — if `addresses` is not an array, the filter should reject it.

Just for the sake of coverage — if we were to remove `f.Array` from that filetr, I don't think there are any unit tests that would fail as a result.

Reply:
np. Added this test case